### PR TITLE
Implement event-driven rendering

### PR DESCRIPTION
**Pull request by eile**

### CODE REVIEW

**Comment by eile** [comment]
@favreau: There is a small chance this breaks VWS. requestFrame does postRedraw(), and I think this should be sufficient to reach the publishFrame(), but I could/did not test it.

**Comment by eile** [comment]
retest this please

**Comment by eile** [comment]
retest this please

**Comment by eile** [comment]
Don't merge. Found a race.

File: livre/core/cache/CacheObjectObserver.h [hunk 1]
```diff
@@ -34,14 +34,6 @@ class CacheObjectObserver
     friend class CacheObject;
 
 protected:
-
-    /**
-     * Notifies the \see CacheObserver before loading.
```
Comment (bilgili) [inline livre/core/cache/CacheObjectObserver.h]: Why did you delete those ?

Comment (eile) [inline livre/core/cache/CacheObjectObserver.h]: Was unused, and it simplified some code I removed later.

**Comment by eile** [comment]
Good to review (and merge if Jenkins agrees)

File: livre/core/cache/CacheObjectObserver.h (hunk 1 above)
Comment (bilgili) [inline livre/core/cache/CacheObjectObserver.h]: Can we take it back, or remove preunload to ? It is not symmetric like this.

**Comment by eile** [comment]
retest this please

File: livre/core/cache/CacheObjectObserver.h (hunk 1 above)
Comment (eile) [inline livre/core/cache/CacheObjectObserver.h]: done.

**Comment by bilgili** [comment]
retest this please

**Comment by bilgili** [comment]
Something is failing on the cache test.

**Comment by eile** [comment]
Yep, no override/final on a virtual :P

**Comment by eile** [comment]
retest this please

**Comment by bilgili** [comment]
retest this please

**Comment by eile** [comment]
So the retest works for you, but not for me. Great. (see https://bbpteam.epfl.ch/project/issues/browse/HELP-3851)

**Comment by bilgili** [comment]
Sometimes you have two do twice, three times etc.
1) Either triggering problem from github
2) Or Jenkins error
